core/evm_ext.go

```diff
@@ -0,0 +1,107 @@
+// (c) 2024, Ava Labs, Inc. All rights reserved.
+// See the file LICENSE for licensing terms.
+
+package core
+
+import (
+	"github.com/ava-labs/avalanchego/snow"
+	"github.com/ava-labs/subnet-evm/constants"
+	"github.com/ava-labs/subnet-evm/core/vm"
+	"github.com/ava-labs/subnet-evm/params"
+	"github.com/ava-labs/subnet-evm/precompile/contract"
+	"github.com/ava-labs/subnet-evm/precompile/contracts/deployerallowlist"
+	"github.com/ava-labs/subnet-evm/precompile/modules"
+	"github.com/ava-labs/subnet-evm/precompile/precompileconfig"
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/log"
+)
+
+var (
+	newEVM = vm.NewEVM
+)
+
+func init() {
+	vm.NewEVM = NewEVM
```

Comment:
This and the above capture of the original of `vm.NewEVM` strike me as a [code smell](https://en.wikipedia.org/wiki/Code_smell). The non-linearity of the logic flow and the "action at a distance" (substituting a global in another package) are anti-patterns that lead to spaghetti code. It will also void all documentation if the implementation can be subbed out in ad hoc ways.

Reply:
What is your suggested alternative?

We can go back to a style that names the new function (or modifying the imports).
But this is problematic since we cannot control all the callsites, especially new ones that may materialize in upstream, which I also noticed in rebasing this PR. See this commit for the original style: https://github.com/ava-labs/subnet-evm/pull/1320/commits/b1bfabb7feb1e07c64010f2c67df3192b9fc54fe

We already configure the VM globally by importing packages and altering global state, such as the precompile module registry.

I agree that global state is up on the list of bad things, but I think it is preferable to callsite modifications.